sips/sip-015/sip-015-network-upgrade.md

```diff
@@ -0,0 +1,1644 @@
+often for the purposes of multiplexing one address into many user accounts.
+Adding a native `stx-transfer-memo?` function makes it easy for smart contracts
+to interact with exchanges that follow this convention.
+
+**Examples:**
+
+```clarity
+(as-contract
+  (stx-transfer? u50 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY tx-sender 0x00)) ;; Returns (err u4)
+(stx-transfer-memo? u60 tx-sender 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY 0x010203)) ;; Returns (ok true)
+```
+
+### New method: `is-standard`
+
+* **Input Signature:** `(is-standard (standard-or-contract principal))`
+* **Output Signature:** `bool`
+
+Tests whether `standard-or-contract` _matches_ the current network
+type, and therefore represents a principal that can spend tokens on the current
+network type. That is, the network is either of type `mainnet`, or `testnet`.
+Only `SPxxxx` and `SMxxxx` _c32check form_ addresses can spend tokens on
+a mainnet, whereas only `STxxxx` and `SNxxxx` _c32check forms_ addresses can spend
+tokens on a testnet. All addresses can _receive_ tokens, but only principal
+_c32check form_ addresses that match the network type can _spend_ tokens on the
+network.  This method will return `true` if and only if the principal matches
+the network type, and false otherwise.
+
+**Rationale:** This method facilitates authenticating principal types supplied
+by contract-calls or from decoded data.  Because principals must have the
+correct version byte in order to spend tokens, ensuring contracts can perform
+this check is of paramount importance to avoid loss-of-funds.
+
+**Examples:**
+
+```clarity
+(is-standard 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6) ;; returns true on testnet and false on mainnet
+(is-standard 'STB44HYPYAT2BB2QE513NSP81HTMYWBJP02HPGK6.foo) ;; returns true on testnet and false on mainnet
+(is-standard 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY) ;; returns true on mainnet and false on testnet
+(is-standard 'SP3X6QWWETNBZWGBK6DRGTR1KX50S74D3433WDGJY.foo) ;; returns true on mainnet and false on testnet
+(is-standard 'SZ2J6ZY48GV1EZ5V2V5RB9MP66SW86PYKKQ9H6DPR) ;; returns false on both mainnet and testnet
+```
+
+### New method: `to-consensus-buff`
+
+* **Input Signature:** `(to-consensus-buff x)`
+* **Output Signature:** `(optional buff))`
+
+The `to-consensus-buff` function is a special function that will serialize any
+Clarity value into a buffer, using the SIP-005 serialization of the
+Clarity value. Not all values can be serialized: some value's
+consensus serialization is too large to fit in a Clarity buffer (this
+is because of the type prefix in the consensus serialization).
+
+If the value cannot fit as serialized into the maximum buffer size,
+this returns `none`, otherwise, it will be
+`(some consensus-serialized-buffer)`. During type checking, the
+analyzed type of the result of this method will be the maximum possible
+consensus buffer length based on the inferred type of the supplied value.
```

Comment:
General rule in Clarity was that there is a hard limit of 1Mb IIRC which is imposed on practically anything (variables, arguments, return values, contract size), so I think it would be worth to mention this limit here as well and make it explicit.
Moreover if it is possible that value which fits within this hard limit can be serialized into a buffer that exceeds it - it should be also explicitly noted here.

Reply:
Added